### PR TITLE
Add io.github.mpobaschnig.Vaults

### PR DESCRIPTION
**Pull request by mpobaschnig**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - "--filesystem=host" needed so the user can create the vault wherever they have access to
  - "--talk-name=org.freedesktop.Flatpak" needed to access gocryptfs/cryfs binaries and fusermount to mount/unmount the directories
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am the maintainer.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.mpobaschnig.Vaults.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        "--filesystem=host"
+    ],
+    "cleanup": [
+        "/include",
+        "/lib/pkgconfig"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
```
Comment (hfiguiere) [inline io.github.mpobaschnig.Vaults.json:25]: this is not available with the builders.

you need to had the crate to the manifest. See https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo for a tool that helps to do so.

File: io.github.mpobaschnig.Vaults.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        ],
+        "test-args" : [
+            "--socket=x11",
+            "--share=network"
```
Comment (hfiguiere) [inline io.github.mpobaschnig.Vaults.json:29]: same as above. no x11, no network.

File: io.github.mpobaschnig.Vaults.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "vaults",
+    "finish-args" : [
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--talk-name=org.a11y.Bus",
+        "--talk-name=org.freedesktop.Flatpak",
+        "--filesystem=host"
```
Comment (hfiguiere) [inline io.github.mpobaschnig.Vaults.json:16]: do you really need "host" access?

File: io.github.mpobaschnig.Vaults.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    "app-id" : "io.github.mpobaschnig.Vaults",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "40",
+    "sdk" : "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "vaults",
+    "finish-args" : [
+        "--socket=fallback-x11",
```
Comment (hfiguiere) [inline io.github.mpobaschnig.Vaults.json:11]: fallback-x11 still requires `--share=ipc`

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: io.github.mpobaschnig.Vaults.json (hunk 1 above)
Comment (bilelmoussaoui) [reply io.github.mpobaschnig.Vaults.json:25]: There's no need to as I believe the app is based on a template I made, you can run `meson _build && ninja -C _build dist` which will create a tarball you can use instead.

File: io.github.mpobaschnig.Vaults.json (hunk 2 above)
Comment (barthalion) [reply io.github.mpobaschnig.Vaults.json:29]: I think anything related to `test` arguments is being ignored by flatpak-builder in the sandbox mode.

File: io.github.mpobaschnig.Vaults.json (hunk 1 above)
Comment (mpobaschnig) [reply io.github.mpobaschnig.Vaults.json]: I'm using a tarball now, thanks!

File: io.github.mpobaschnig.Vaults.json (hunk 2 above)
Comment (mpobaschnig) [reply io.github.mpobaschnig.Vaults.json]: Removed.

File: io.github.mpobaschnig.Vaults.json (hunk 3 above)
Comment (mpobaschnig) [reply io.github.mpobaschnig.Vaults.json:16]: I need this so user can access e.g. USB drives under /run or other locations like /mnt where they have access to. I'm not sure if I should set "home" access instead and additionally enable specific directories, but I don't know which directories users might come up with.

File: io.github.mpobaschnig.Vaults.json (hunk 4 above)
Comment (mpobaschnig) [reply io.github.mpobaschnig.Vaults.json:11]: Added.

File: io.github.mpobaschnig.Vaults.json [hunk 5]
```diff
@@ -21,18 +21,33 @@
...
+            "name" : "sassc",
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/lazka/sassc.git",
+                    "commit" : "82803377c33247265d779af034eceb5949e78354"
+                }
+            ]
+        },
```
Comment (bilelmoussaoui) [inline io.github.mpobaschnig.Vaults.json:50]: Both these are build dependencies of libadwaita, you should add `"cleanup": ["*"]` to both of them to remove them completely post-build

Comment (mpobaschnig) [reply io.github.mpobaschnig.Vaults.json]: Added.

**Comment by bilelmoussaoui** [comment]
bot, build io.github.mpobaschnig.Vaults

**Comment by mpobaschnig** [comment]
Sorry, forgot to put the raw link into screenshot section. I'll fix it quickly.

**Comment by mpobaschnig** [comment]
Should be fixed now.

**Comment by bilelmoussaoui** [comment]
bot, build io.github.mpobaschnig.Vaults

**Review by barthalion — COMMENTED** [review]
Appdata should mention that user should install gocryptfs or cryfs to use the app.

**Comment by mpobaschnig** [comment]
Added clarification in the description that the user needs to install gocrytpfs/cryfs/fuse for this app to work.

**Comment by mpobaschnig** [comment]
Thanks! :)

**Pull request closed without merging**